cfme/infrastructure/deployment_roles.py

```diff
@@ -1,67 +1,249 @@
 """ A model of an Infrastructure Deployment roles in CFME"""
 
 from functools import partial
-from navmazing import NavigateToAttribute, NavigateToSibling
-from selenium.common.exceptions import NoSuchElementException
-
-from cfme.exceptions import ListAccordionLinkNotFound
-from cfme.fixtures import pytest_selenium as sel
-from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
+from navmazing import NavigateToSibling, NavigateToAttribute, NavigateToObject
+from widgetastic.exceptions import NoSuchElementException
+from cfme.exceptions import RoleNotFound
+from widgetastic.widget import View, ParametrizedView
+from widgetastic_patternfly import BootstrapNav, Button, Dropdown, FlashMessages
+from widgetastic_manageiq import (
+    Accordion, BootstrapSelect, BreadCrumb, ItemsToolBarViewSelector, PaginationPane,
+    SummaryTable, Table, Text, BootstrapTreeview, BaseTileIconEntity, BaseQuadIconEntity,
+    BaseListEntity, JSBaseEntity, NonJSBaseEntity, BaseEntitiesView)
+from widgetastic.utils import Version, VersionPick
+from cfme.base.ui import BaseLoggedInPage
 from utils.appliance import Navigatable
-from utils.appliance.implementations.ui import (CFMENavigateStep, navigate_to,
-                                                navigator)
-from cfme.web_ui import (listaccordion as list_acc, match_location, Quadicon,
-                         Region, toolbar as tb)
-from utils import version
-from utils.pretty import Pretty
+from utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
+from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
+from cfme.common.provider_views import InfraProviderDetailsView
```

Comment:
**REQUIRED**
Please follow [style guide](http://cfme-tests.readthedocs.io/guides/dev_guide.html) for imports .